extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ByteChunkReader.java

```diff
@@ -93,30 +66,15 @@ public WritableByteChunk<Values> readChunk(Iterator<ChunkInputStreamGenerator.Fi
 
         final int numValidityLongs = options.useDeephavenNulls() ? 0 : (nodeInfo.numElements + 63) / 64;
         try (final WritableLongChunk<Values> isValid = WritableLongChunk.makeWritableChunk(numValidityLongs)) {
-            if (options.useDeephavenNulls() && validityBuffer != 0) {
-                throw new IllegalStateException("validity buffer is non-empty, but is unnecessary");
-            }
-            int jj = 0;
-            for (; jj < Math.min(numValidityLongs, validityBuffer / 8); ++jj) {
-                isValid.set(jj, is.readLong());
-            }
-            final long valBufRead = jj * 8L;
-            if (valBufRead < validityBuffer) {
-                is.skipBytes(LongSizedDataStructure.intSize(DEBUG_NAME, validityBuffer - valBufRead));
-            }
-            // we support short validity buffers
-            for (; jj < numValidityLongs; ++jj) {
-                isValid.set(jj, -1); // -1 is bit-wise representation of all ones
-            }
-            // consumed entire validity buffer by here
+            readValidityBuffer(is, numValidityLongs, validityBuffer, isValid, DEBUG_NAME);
 
             final long payloadRead = (long) nodeInfo.numElements * Byte.BYTES;
             Assert.geq(payloadBuffer, "payloadBuffer", payloadRead, "payloadRead");
 
             if (options.useDeephavenNulls()) {
-                useDeephavenNulls(conversion, is, nodeInfo, chunk, outOffset);
+                useDeephavenNulls(is, nodeInfo, chunk, outOffset);
```

Comment:
In these replicated readers and in the default reader, it seems the useDeephavenNulls option isn't exercised.